### PR TITLE
Modal listing click

### PR DESCRIPTION
**Pull request by schwers**
Depends on https://github.com/reddit/reddit-mobile/pull/1076

### CODE REVIEW

**Comment by schwers** [comment]
Link to the ticket https://reddit.atlassian.net/browse/CHAN-511 (staging link and mocks in the there)

For testing locally, you can add this config to `reducers/accounts.js`

```js
      Object.keys(accounts).forEach(accountName => {
        const account = accounts[accountName];
        accounts[accountName] = account.set({
          features: {
            ...account.features,
            mweb_xpromo_modal_listing_click_ios: {
              owner: 'channels',
              experiment_id: 165,
              variant: 'daily_dismissible',
            },
            mweb_xpromo_modal_listing_click_android: {
              owner: 'channels',
              experiment_id: 165,
              variant: 'daily_dismissible',
            },
          },
        });
      });
```

File: src/app/components/XPromoListingClickModal/styles.less [hunk 1]
```diff
@@ -0,0 +1,100 @@
+@import (reference) '~app/less/variables';
+@import (reference) '~app/less/mixins/importAll';
+@import (reference) '~app/less/themes/themeify';
+
+
+.backgroundImage() {
+  background-image: url('/img/listing-click-pattern.png');
```
Comment (tomsoir) [inline src/app/components/XPromoListingClickModal/styles.less]: Oh i c. there is a night mode also. :) 👍

File: src/app/components/XPromoWrapper/index.jsx [hunk 2]
```diff
@@ -55,9 +55,6 @@ class XPromoWrapper extends React.Component {
   }
 
   componentDidMount() {
-    // Indicate that we've displayed a crosspromotional UI, 
-    // so we don't keep showing them during this browsing session.
-    this.props.recordXPromoShown();
```
Comment (tomsoir) [inline src/app/components/XPromoWrapper/index.jsx]: If there is no needs in 'recordXPromoShown' anymore, I believe we can delete next code at the top of this file too :)
...
const T = React.PropTypes;
... 
static propTypes = {
    recordXPromoShown: T.func.isRequired,
};
...

File: src/app/constants.js [hunk 3]
```diff
@@ -164,19 +169,25 @@ export const genericErrors = {
...
 export const EVERY_WEEK = 'every_week';
 export const EVERY_TWO_WEEKS = 'every_two_weeks';
 
-export const experimentFrequencyVariants = {};
-experimentFrequencyVariants[EVERY_DAY] = 24 * 60 * 60 * 1000;
-experimentFrequencyVariants[EVERY_THREE_DAYS] = 3 * 24 * 60 * 60 * 1000;
-experimentFrequencyVariants[EVERY_WEEK] = 1 * 7 * 24 * 60 * 60 * 1000;
-experimentFrequencyVariants[EVERY_TWO_WEEKS] = 2 * 7 * 24 * 60 * 60 * 1000;
+const HOUR_IN_MS = 60 * 60 * 1000;
+
```
Comment (tomsoir) [inline src/app/constants.js]: 👍

File: src/app/reducers/xpromo/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ */
+import { combineReducers } from 'redux';
+
+import interstitials from './interstitials';
+import listingClick from './listingClick';
+
+export default combineReducers({
+  interstitials,
+  listingClick,
+});
```
Comment (tomsoir) [inline src/app/reducers/xpromo/index.js]: This is great 👍 I made such think in "Persistent Footer" task too. Will return it back and wait for rebase.

File: src/lib/xpromoState.js [hunk 5]
```diff
@@ -131,10 +133,10 @@ export function getXPromoLink(state, path, linkType, additionalData={}) {
 }
 
 function getClosingTimeRange(state) {
-  const defaultRange = frequency[EVERY_TWO_WEEKS];
+  const defaultRange = FREQUENCIES[EVERY_TWO_WEEKS];
   const experimentData = getFrequencyExperimentData(state);
   if (experimentData) {
-    return (frequency[experimentData.variant] || defaultRange);
+    return (FREQUENCIES[experimentData.variant] || defaultRange);
```
Comment (tomsoir) [inline src/lib/xpromoState.js]: Thank you 👍

File: src/app/components/XPromoListingClickModal/styles.less (hunk 1 above)
Comment (schwers) [inline src/app/components/XPromoListingClickModal/styles.less]: Yeah its unfortunately that its a little large. The preload divs should help to fetch it before its needed. (Plus its most likely already cached from the previous experiment)

File: src/app/components/XPromoListingClickModal/styles.less [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+.promoImage() {
+  background-image: url('/img/tutorial-app.png');
+  background-size: 100%;
+  background-position: center;
+  background-repeat: no-repeat;
+}
+
+.XPromoListingClickModal {
+  .ModalWrapper();
```
Comment (schwers) [inline src/app/components/XPromoListingClickModal/styles.less]: Ah, good catch and a simple fix.  👍

File: src/app/components/XPromoWrapper/index.jsx (hunk 2 above)
Comment (schwers) [inline src/app/components/XPromoWrapper/index.jsx]: 👍  good catch

File: src/app/components/XPromoListingClickModal/styles.less (hunk 6 above)
Comment (schwers) [inline src/app/components/XPromoListingClickModal/styles.less]: Fixed in https://github.com/reddit/reddit-mobile/pull/1081/commits/ae890ef0e04db44b68967c8df504b7e43efed6a0. Adds a responsive square mixin and additionally resizes the image for landscape

**Comment by schwers** [comment]
💇‍♂️

**Comment by tomsoir** [comment]
🐟 good to go.

**Comment by schwers** [comment]
Added some event fixups (https://github.com/reddit/reddit-mobile/pull/1081/commits/03204d3d329d981d45ac1d8e90d12ef576fcafce), and pushed the rebase.

💇

**Pull request merged**